src/wallet/coinselection.cpp

```diff
@@ -395,3 +387,30 @@ CAmount GetSelectionWaste(const std::set<CInputCoin>& inputs, CAmount change_cos
 
     return waste;
 }
+
+CAmount SelectionResult::GetSelectedValue() const
+{
+    CAmount ret = 0;
+    for (const auto& coin : m_selected_inputs) {
+        ret += coin.txout.nValue;
+    }
+    return ret;
+}
+
+void SelectionResult::Clear()
+{
+    m_selected_inputs.clear();
+}
+
+void SelectionResult::AddInput(const OutputGroup& group)
+{
+    util::insert(m_selected_inputs, group.m_outputs);
+    m_use_effective = !group.m_subtract_fee_outputs;
+}
+
+std::vector<CInputCoin> SelectionResult::GetInputVector() const
+{
+    std::vector<CInputCoin> coins(m_selected_inputs.begin(), m_selected_inputs.end());
+    Shuffle(coins.begin(), coins.end(), FastRandomContext());
```

Comment:
I think randomizing in a `Get` function is very unexpected. You wouldn't expect it to return a different value every time. I'd prefer to either rename this function to e.g. `RandomizedInputVector()` or, do it at the call site.

Reply:
I've added `GetInputSet` to just return `m_selected_inputs` and changed this to be `GetShuffledInputVector`.

Reply:
Thanks!